docs/content/basics/ui_testing.md

```diff
@@ -1,4 +1,148 @@
 # UI testing
 
-To be done
+UI tests are a part of instrumentation tests. <br/>
+That's why everything from [Instrumented tests topic](https://android-ui-testing.github.io/Cookbook/basics/instrumented_testing_basics/) is applicable to UI testing.
 
+Ui testing goal is to check some scenarios of using your application.<br/>
+Usually, we do ui testing for some common users scenario:
+
+* Login to your application
+* Process of ordering 
+* Creating new chat 
+* And so on
+
+Simple Ui test can't catch the problem with wrong padding or wrong color of your view (only in cases where you specially check that) - because there is another type of tests used to catch such problems called `Screenshot testing` - you can read [Screenshot testing article](https://android-ui-testing.github.io/Cookbook/basics/screenshot_testing/) to get familiar with that.<br/>
+
+So, if once we decided that we need to check users scenarios - we should definitely start writing Ui tests and let's start with tools, that will help us to write this type of tests.
+
+## Main tool - Espresso and UiAutomator
```

Comment:
Maybe worth mentioning that Jetpack Compose testing library for Composables?

"Different testing tools. Espresso and UIAutomator can still test a Compose Layout - searching by text, resource, etc. However, they don’t have access to Composables’ semantics and can’t fully test them. Therefore, it’s recommended to use the Jetpack Compose testing library as it can access semantics and fully test Composables on the screen."
Link:
https://dmytroshuba.com/blog/jetpack-compose-ui-testing/